plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergSnapshots.java

```diff
@@ -0,0 +1,97 @@
+
+import static io.trino.plugin.iceberg.IcebergQueryRunner.createIcebergQueryRunner;
+import static io.trino.testing.sql.TestTable.randomTableSuffix;
+import static java.lang.String.format;
+
+public class TestIcebergSnapshots
+        extends AbstractTestQueryFramework
+{
+    public static final int PARENT_ID_FIELD = 1;
+    public static final int ID_FIELD = 0;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        return createIcebergQueryRunner(ImmutableMap.of(), new IcebergConfig().getFileFormat(), ImmutableList.of());
+    }
+
+    @Test
+    public void testReadingFromSpecificSnapshot()
+    {
+        String tableName = "test_reading_snapshot" + randomTableSuffix();
+        assertUpdate(format("CREATE TABLE %s (a bigint, b bigint)", tableName));
+        assertUpdate(format("INSERT INTO %s VALUES(1, 1)", tableName), 1);
+        List<MaterializedRow> materializedRows = getQueryRunner().execute(
+                format("SELECT snapshot_id, parent_id FROM \"%s$snapshots\" ORDER BY snapshot_id", tableName)).getMaterializedRows();
+        MaterializedRow oldestSnapshot = getOldestSnapshot(materializedRows);
+        long emptyTableSnapshot = (long) oldestSnapshot.getField(ID_FIELD);
+        long afterFirstInsertSnapshot = (long) getNextSnapshot(materializedRows, emptyTableSnapshot).getField(ID_FIELD);
+
+        assertQuery(format("SELECT count(*) FROM \"%s@%d\"", tableName, emptyTableSnapshot), "VALUES(0)");
+        assertQuery(format("SELECT * FROM \"%s@%d\"", tableName, afterFirstInsertSnapshot), "VALUES(1,1)");
+        assertUpdate(format("DROP TABLE %s", tableName));
+    }
+
+    @Test
+    public void testSelectWithMoreThanOneSnapshotOfTheSameTable()
+    {
+        String tableName = "test_reading_snapshot" + randomTableSuffix();
+        assertUpdate(format("CREATE TABLE %s (a bigint, b bigint)", tableName));
+        assertUpdate(format("INSERT INTO %s VALUES(1, 1)", tableName), 1);
+        assertUpdate(format("INSERT INTO %s VALUES(2, 2)", tableName), 1);
+        assertUpdate(format("INSERT INTO %s VALUES(3, 3)", tableName), 1);
+        List<MaterializedRow> materializedRows = getQueryRunner().execute(
+                format("SELECT snapshot_id, parent_id FROM \"%s$snapshots\" ORDER BY snapshot_id", tableName)).getMaterializedRows();
+        MaterializedRow oldestSnapshot = getOldestSnapshot(materializedRows);
+        long emptyTableSnapshot = (long) oldestSnapshot.getField(ID_FIELD);
+        long afterFirstInsertSnapshot = (long) getNextSnapshot(materializedRows, emptyTableSnapshot).getField(ID_FIELD);
+        long afterSecondInsertSnapshot = (long) getNextSnapshot(materializedRows, afterFirstInsertSnapshot).getField(ID_FIELD);
+
+        assertQuery(format("SELECT * FROM %s", tableName), "SELECT * FROM (VALUES(1,1), (2,2), (3,3))");
+        assertQuery(format("SELECT * FROM %s EXCEPT (SELECT * FROM \"%s@%d\" EXCEPT SELECT * FROM \"%s@%d\")",
```

Comment:
nit: you can use argument index in format string to avoid passing `tableName` multiple times in args.